src/get_maintainer_wrapper.sh

```diff
@@ -47,40 +48,168 @@ function print_files_authors()
+}
+
+# Checks if the given path is a patch file. Argument is the path
+function is_a_patch
+{
+  local -r FILE_PATH="$@"
+  if [[ ! -f "$FILE_PATH" ]]; then
+    return 1
+  fi
+
+	# The following array stores strings that are expected to be present
+	# in a patch file. The absence of any of these strings makes the
+	# given file be considered NOT a patch
+	local -ar PATCH_EXPECTED_STRINGS=(
+		"diff --git"
+		"Subject:"
+		"Date:"
+		"From:"
+		"deletions(-)"
+		"insertions(+)"
+	)
+
+	for expected_str in "${PATCH_EXPECTED_STRINGS[@]}"; do
+		if ! grep -F "$expected_str" "$FILE_PATH" &> /dev/null; then
+			return 1
+		fi
+	done
+
+	return 0
+}
+
+# Executes get_maintainer with the given file or dir. Also accepts the
+# parameter "--authors" or "-a" before the file/dir, to enable authors
+# printing.
 function execute_get_maintainer()
 {
-  local FILE_OR_DIR_CHECK
+  local FILE_OR_DIR
   local print_authors
+	local is_file_a_patch=true
+	local is_file_inside_kernel_tree=true
 
   local -r script="scripts/get_maintainer.pl"
-  local -r options="--separator , --nokeywords --nogit --nogit-fallback --norolestats "
-  local -r getmaintainers="perl $script $options"
+  local options="--separator , --nokeywords --nogit --nogit-fallback --norolestats "
 
-  # Check if the command was invoked from the kernel root tree
-  if [ ! -f $script ]; then
-    complain "You have to execute this command from the linux tree"
-    return 1
-  fi
+  local -r original_working_dir=$PWD
+  local kernel_root=""
+  local path_from_kernel_root=""
 
+  # Check function options
   if [[ $# -ge 1 && ( $1  = "--authors" || $1 = "-a" ) ]]; then
-      FILE_OR_DIR_CHECK=$2
+      FILE_OR_DIR=$2
       print_authors=true
   else
-      FILE_OR_DIR_CHECK=$1
+      FILE_OR_DIR=$1
       print_authors=false
   fi
 
+  # If no file is given, assume "."
+  if [ -z $FILE_OR_DIR ]; then
+    FILE_OR_DIR="."
+  fi
+  FILE_OR_DIR="$(realpath $FILE_OR_DIR)"
+
   # Check if is a valid path
-  if [ ! -d $FILE_OR_DIR_CHECK -a ! -f $FILE_OR_DIR_CHECK ]; then
+  if [ ! -d $FILE_OR_DIR -a ! -f $FILE_OR_DIR ]; then
     complain "Invalid path"
     return 1
   fi
 
+	# if given path is not a patchfile, add -f to get_maintainer.pl options
+	if ! is_a_patch "$FILE_OR_DIR"; then
+		is_file_a_patch=false
+		options="$options -f "
+	fi
+
+  # try to find kernel root at given path
+  kernel_root="$(find_kernel_root $FILE_OR_DIR)"
+  if [ -z "$kernel_root" ]; then
+		is_file_inside_kernel_tree=false
+		# fallback: try to find kernel root at working path
+    kernel_root="$(find_kernel_root $original_working_dir)"
+  fi
+  # Check if kernel root was found.
+  if [ -z "$kernel_root" ]; then
+		complain "Neither the given path nor the working path is in a kernel tree."
+		return 1
+  fi
+
+	# If file is not a patch and outside a kernel tree, it must be an user's
+	# mistake. Although get_maintainer.pl can handle this, and would print
+	# the kernel root maintainer, linux torvalds, it's better to abort
```

Comment:
I did not understood what you meant by "linux torvalds", but in any case, I prefer no name references in the code.